app/models/application.rb

```diff
@@ -14,6 +14,17 @@ class Application < ApplicationRecord
 
   validate :source_must_be_compatible
 
+  def remove_availability_status!
+    remove_availability_status
+    save
```

Comment:
I would use `save!` instead, that way it raises on a failure, I think it fits better with the method name since it ends in `!`

edit: here/endpoint etc.